stackslib/src/chainstate/stacks/db/headers.rs

```diff
@@ -111,19 +112,20 @@ impl FromRow<StacksMicroblockHeader> for StacksMicroblockHeader {
 impl StacksChainState {
     /// Insert a block header that is paired with an already-existing block commit and snapshot
     pub fn insert_stacks_block_header(
-        tx: &mut DBTx,
+        tx: &DBTx,
         parent_id: &StacksBlockId,
         tip_info: &StacksHeaderInfo,
         anchored_block_cost: &ExecutionCost,
         affirmation_weight: u64,
     ) -> Result<(), Error> {
-        assert_eq!(
-            tip_info.stacks_block_height,
-            tip_info.anchored_header.total_work.work
-        );
+        let header = match tip_info.anchored_header {
+            StacksBlockHeaderTypes::Epoch2(ref x) => x,
+            _ => return Err(Error::InvalidChildOfNakomotoBlock),
+        };
```

Comment:
Could be simplified with `let-else`:

```suggestion
        let StacksBlockHeaderTypes::Epoch2(header) = &tip_info.anchored_header else {
            return Err(Error::InvalidChildOfNakomotoBlock)
        };
```